source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1358,85 +1358,66 @@ It will be refreshed by the next periodic check or,
 if an application operation needs the server sooner than that,
 then a re-check will be triggered by the server selection algorithm.
 
-"not master" and "node is recovering"
-`````````````````````````````````````
+Transient Server Errors
+```````````````````````
 
-These errors are detected from a getLastError response,
-write command response, or query response. Clients MUST consider a server
-error to be a "node is recovering" error if the substrings "node is recovering"
-or "not master or secondary" are anywhere in the error message.
-Otherwise, if the substring "not master" is in the error message it is a
-"not master" error::
-
-    def is_recovering(message):
-        return ("not master or secondary" in message
-            or "node is recovering" in message)
-
-    def is_notmaster(message):
-        if is_recovering(message):
-            return false
-        return ("not master" in message)
```

Comment:
It's possible for old versions (3.0 and possibly 3.2) of the server to return a "not master" error message without returning the corresponding error code. We should re-consider entirely removing this part.

Reply:
Alas this section is still very much alive. E.g., an insert on a secondary running 3.0:
```
db.runCommand({insert:'collection', documents: [{}]})
{ "note" : "from execCommand", "ok" : 0, "errmsg" : "not master" }
```
There is *no* error code whatsoever!

Reply:
In that case, I can change this to first check if the returned error has a code. If no error code exists, we'll do the same as before, checking the message for a substring.

If so, I'll update and add this section back, and likely update the resumable/retryable error explanations to again refer back to this section (just with a different list of error codes).

Does that sound alright?

Reply:
Yup!